internal/pkg/fieldtags/analyzer.go

```diff
@@ -149,30 +148,7 @@ func (sp *sourcePatterns) isSource(field *ast.Field) bool {
 
 // IsSource determines whether a FieldAddr is a source, that is whether it refers to a field previously identified as a source.
 func (t TaggedFields) IsSource(f *ssa.FieldAddr) bool {
-	n, ok := namedType(f)
-	if !ok {
-		return false
-	}
-	fName := utils.FieldName(f)
-	for _, tf := range t {
-		if tf.matches(n.String(), fName) {
-			return true
-		}
-	}
-	return false
-}
-
-func namedType(f *ssa.FieldAddr) (*types.Named, bool) {
-	structType := f.X.Type()
-	deref := utils.Dereference(structType)
-	named, ok := deref.(*types.Named)
-	return named, ok
-}
-
-func (t TaggedField) matches(typeName, fieldName string) bool {
-	return fmt.Sprintf("%s.%s", t.pkg, t.typ) == typeName && t.field == fieldName
-}
-
-func (t TaggedField) String() string {
-	return fmt.Sprintf("%s.%s.%s", t.pkg, t.typ, t.field)
+	// incantation plundered from the ssa package
+	field := f.X.Type().Underlying().(*types.Pointer).Elem().Underlying().(*types.Struct).Field(f.Field)
```

Comment:
This seems prone to panic.  Are the assumptions in our use-case the same as those from wherever you plundered this?

Reply:
I think they are. The way it's documented seems to suggest that this is the canonical way to get the field, so I would expect it not to panic.

I plundered this from the documentation for `ssa.FieldAddr`. I've made the comment more specific in that regard. I could extract this to a function that checks every step of the way to ensure a panic cannot occur.